### ISSUE TITLE
upgrading from 0.14.0 to 0.14.1 is a breaking change

### ISSUE THREAD

**Issue by swiknaba** [nesquena/rabl#724]
Having the following setup in an older app running on `ruby 2.4.6` and `rails 4.2.11.1`

```ruby
# index.json.rabl
node :foo do
  partial('foos/list', object: @foos)
end
```

```ruby
# list.json.rabl
collection @foos, root: false, object_root: false
extends('foos/show')
```

```ruby
# show.json.rabl
object @foo => nil
attributes :id, :name
```

```ruby
# foos_controller.rb
def index
  @foos = Foo.all
  render 'foos/index'
end
```

With Rabl `0.14.0` this renders properly:
```json
{
    "foos": [
        { "id": 1, "name": "foo one"},
        { "id": 2, "name": "foo two"},
    ]
}
```

Upgrading to `0.14.1` it just renders nothing, no error raised:
```json
{
    "foos": {}
}
```

I am not very familiar with Rabl, to be honest, are the views here just basically wrong and working by chance in this setup, or did we find a breaking change that slipped in a minor version bump?

**Comment by nesquena** [comment]
I’ll look into this soon, sorry to hear about that. Once I can confirm, I’ll yank 0.14.1 until this is resolved

**Comment by swiknaba** [comment]
Thanks! I create a sample app to reproduce the issue: https://github.com/swiknaba/rabl_0140

**Comment by dmke** [comment]
Same problem here.

We're presenting an `ActiveRecord::Relation` object in our templates:

```ruby
# controller
@tags = Tag.where(...)

# template
object @tags
attributes :id, :name, :position
```

The workaround for this is converting the collection to an array (but that's obviously bad for performance):

```ruby
object @tags.to_a
```

**Comment by dramalho** [comment]
Yeah, I just bitten by this, locking the gem to 0.14.0 until either it's fixed or, presumably, I move to Rails > 4.2 and this is no longer an issue

**Comment by tagliala** [comment]
Confirm this issue.

Problem is in commit: 4cf2c667a8f05eb5e04abf371c0978f6604574ab

exactly this change: https://github.com/nesquena/rabl/commit/4cf2c667a8f05eb5e04abf371c0978f6604574ab#diff-5a71730235e40573daff6c167a927d5dR85-R89

because on Rails 4 an active record relation responds to both `:map` and `:each` but it is not an enumerable

```
[2] pry(main)> Document.all.respond_to?(:map)
=> true
[3] pry(main)> Document.all.respond_to?(:each)
=> true
[4] pry(main)> Document.all.is_a?(Enumerable)
=> false
```

Hope it helps

**Comment by nesquena** [comment]
Sorry to hear about this, I am no longer actively maintaining RABL unfortunately as you probably gathered, though I still use an older version on several apps.

If anyone can point me to a PR, and a recommended course of action, I'd be more than happy to push a 0.14.2 to resolve this

**Comment by nesquena** [comment]
@tagliala Sent me a PR, merged and released as 0.14.3. I've also marked you as a contributor @tagliala if you can help keep RABL alive and well in whatever capacity you can, I appreciate it. That goes for anyone else reading this thread too.

**Comment by tagliala** [comment]
@nesquena apologies are not necessary. Thank you for all these years, I've been using rabl since forever 🙏🏼
